server/tools/jgroups.sh

```diff
@@ -10,6 +10,7 @@ if [ -n "$JGROUPS_DISCOVERY_PROTOCOL" ]; then
     echo "Setting JGroups discovery to $JGROUPS_DISCOVERY_PROTOCOL with properties $JGROUPS_DISCOVERY_PROPERTIES_PARSED"
     echo "set keycloak_jgroups_discovery_protocol=${JGROUPS_DISCOVERY_PROTOCOL}" >> "$JBOSS_HOME/bin/.jbossclirc"
     echo "set keycloak_jgroups_discovery_protocol_properties=${JGROUPS_DISCOVERY_PROPERTIES_PARSED}" >> "$JBOSS_HOME/bin/.jbossclirc"
+    echo "set keycloak_jgroups_transport_stack=${JGROUPS_TRANSPORT_STACK:-udp}" >> "$JBOSS_HOME/bin/.jbossclirc"
```

Comment:
I'm just thinking whether or not we should TCP as the default stack.... Most of the folks with need it for AWS or GCP deployments.

WDYT?

Reply:
That to me seems orthogonal to this original issue, which is "allowing to change the stack".

Reply:
But to clarify: there seems to be a major performance gain in certain cases we have tested with TCP_NIO2, however, caused certain issues and had to be reverted. However, if you just wait a little longer until we can stabilize TCP_NIO2, you will get the switch out of box. See https://issues.jboss.org/browse/WFLY-8066

Reply:
My original motivation was to allow folks using this image right away in AWS and GCP. In those deployment scenarios, they would need to switch it to TCP.

But when I think about it, it breaks backwards compatibility. If anyone used our image on prem, he'd need to add a `JGROUPS_TRANSPORT_STACK` environmental variable, whereas the new deployments on AWS would need to use it. So let's keep it as is and maybe someday we'll consider changing the default to TCP.

Reply:
After doing a some research and talking with @stianst, I believe the best option is to use TCP by default. Could we incorporate that change into this PR?

Reply:
Sure, we just need to make sure the new behaviour is mentioned in the release notes and documentation.